_drafts/making-backend-scalable.markdown

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "Making Autolab's Backend Scalable"
+author: Ilter Canberk
+
+tags: [tango, redis, scalability]
+
+---
+
+[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files.
```

Comment:
- adds them __to__ a job queue
  - preposition usage
- shepherds the job __through__ the process
  - Typo
- Autolab has grown to host classes like Distributed Systems, Machine Learning, __and Storage Systems--classes with significantly higher compute requirements.__
  - Lists of three sound better, and joining the two sentences (either with a dash or a semicolon) will improve readability